Crashes on initialization

Just tried to use this and getting this crash every time

I'm on MacOS (arm) with Zed v0.209.7 and using tsgo `Version 7.0.0-dev.20251029.1` based on the path of the `Server Info` block which I then ran `--version` against.

```
configuration: [<nil>], []interface {}


[08:18:44.630] Processed 1 file changes in 8.209µs
panic: ScriptKind must be specified when parsing source files.

goroutine 13306 [running]:
github.com/microsoft/typescript-go/internal/parser.(*Parser).initializeState(0x105e5fb20?, {{0x140009e75e0, 0x42}, {0x14016d5e5a0, 0x42}, {0x0, 0x0, 0x1, 0x1}, {0x0, ...}, ...}, ...)
	github.com/microsoft/typescript-go/internal/parser/parser.go:210 +0x374
github.com/microsoft/typescript-go/internal/parser.ParseSourceFile({{0x140009e75e0, 0x42}, {0x14016d5e5a0, 0x42}, {0x0, 0x0, 0x1, 0x1}, {0x0, 0x0}, ...}, ...)
	github.com/microsoft/typescript-go/internal/parser/parser.go:115 +0xd4
github.com/microsoft/typescript-go/internal/project.(*ParseCache).Acquire(0x105e5d6b0?, {0x10d49d1f0, 0x140003282a0}, {{0x140009e75e0, 0x42}, {0x14016d5e5a0, 0x42}, {0x0, 0x0, 0x1, ...}, ...}, ...)
	github.com/microsoft/typescript-go/internal/project/parsecache.go:56 +0x1b8
github.com/microsoft/typescript-go/internal/project.(*compilerHost).GetSourceFile(0x14000616540, {{0x140009e75e0, 0x42}, {0x14016d5e5a0, 0x42}, {0x0, 0x0, 0x1, 0x1}, {0x0, ...}, ...})
	github.com/microsoft/typescript-go/internal/project/compilerhost.go:128 +0x110
github.com/microsoft/typescript-go/internal/compiler.(*fileLoader).parseSourceFile(0x14000176008, 0x140015cc000)
	github.com/microsoft/typescript-go/internal/compiler/fileloader.go:422 +0x150
github.com/microsoft/typescript-go/internal/compiler.(*parseTask).load(0x140015cc000, 0x14000176008)
	github.com/microsoft/typescript-go/internal/compiler/filesparser.go:77 +0x158
github.com/microsoft/typescript-go/internal/compiler.(*filesParser).start.func1()
	github.com/microsoft/typescript-go/internal/compiler/filesparser.go:220 +0x170
github.com/microsoft/typescript-go/internal/core.(*parallelWorkGroup).Queue.func1()
	github.com/microsoft/typescript-go/internal/core/workgroup.go:42 +0x50
created by github.com/microsoft/typescript-go/internal/core.(*parallelWorkGroup).Queue in goroutine 33
	github.com/microsoft/typescript-go/internal/core/workgroup.go:40 +0x7c

```

My `settings.json` has this although I did try to just use `TypeScript` and that was also failing:
```
  "languages": {
    "TypeScript": {
      "language_servers": ["tsgo", "eslint"]
    },
    "TSX": {
      "language_servers": ["tsgo", "eslint"]
    },
    "JavaScript": {
      "language_servers": ["tsgo", "eslint"]
    }
  },

```

--- a/extension.toml
+++ b/extension.toml
@@ -9,4 +9,11 @@ repository = "https://github.com/zed-extensions/tsgo"
 
 [language_servers.tsgo]
 name = "tsgo"
+# TSGo determines ScriptKind from LSP languageId (textDocument/didOpen).
+# ScriptKind enum definition: https://github.com/microsoft/typescript-go/blob/main/internal/parser/parser.go (search for "type ScriptKind")
+# Panic location (initializeState): https://github.com/microsoft/typescript-go/blob/main/internal/parser/parser.go#L210
+# Maps Zed's language names to standard LSP languageId values:
+#   typescript -> ScriptKindTS, typescriptreact -> ScriptKindTSX,
+#   javascript -> ScriptKindJS, javascriptreact -> ScriptKindJSX
+language_ids = { TypeScript = "typescript", TSX = "typescriptreact", JavaScript = "javascript", JSX = "javascriptreact" }
 languages = ["TypeScript", "TSX", "JavaScript", "JSX"]

